config.js

```diff
@@ -0,0 +1,16 @@
+(function() {
+  'use strict';
+
+  angular.module('esn.auth').constant('config', {
```

Comment:
Do you really want to stay with an angular module and a constant?

If yes:
- you will have to generate the angular-like file which is not really a nice thing
- You do not haver to name the constant `config` since it is probably already used and there will be a name collision. I suggest `esnAuthConfig`

The way we did things in Vue apps were:

1. Have a `openpaas.js` file in `public/env/` folder which put some data in the window object (https://github.com/linagora/openpaas-dashboard/blob/master/public/env/openpaas.js)
2. Load this file in index.html (https://github.com/linagora/openpaas-dashboard/blob/master/public/index.html#L16)
3. Override `process.env` with these values (https://github.com/linagora/openpaas-dashboard/blob/master/src/services/application-settings.js#L29)
4. Mount Docker volume (https://github.com/linagora/openpaas-dashboard/blob/master/Dockerfile#L13)

I am not against the current approach, we just have to be agree and consistant.

@tanandy

Reply:
4. bind mount is sufficient in that case, no need to create a specific volume for persistence

Reply:
btw, i wait the point of view of fmartin

Reply:
Ok, so I like the way this solution makes simpler config file:
```js
window.esnAuthConfig = {
  ...
}
```
instead of
```js
(function() {
  'use strict';

  angular.module('esn.auth').constant('esnAuthConfig', {
    ...
  });
})();
```
And we can add an AngularJS getter in the esn-frontend-common-libs project I guess:
```js
angular.module('esn.auth').value('esnAuthConfig', function($window) {
  return $window.esnAuthConfig;
});
```
The best solution would be to have a config file containing only the configuration and not its loading mecanism
```js
{
  ...
}
```
but I don't know if it is possible and if it really has added value.

Reply:
so we have the default file here : https://github.com/linagora/linshare-ui-user/blob/master/app/scripts/default.config.js

Reply:
and the configuration file that can be override by configmap : https://github.com/linagora/linshare-ui-user/blob/master/app/config/config.js

Reply:
this second file contains only properties needed by end users and you do not need to provide them all, only the ones you need and it will be merged with default.config.js

Reply:
Indeed, good idea to make all products work the same way ! :+1:

Reply:
So finally I used the `env/openpaas.js` initially made for this.